app/features/build_runner/build_runner.rb

```diff
@@ -129,6 +132,52 @@ def checkout_sha
     def pre_run_action
       logger.debug("Running pre_run_action in checkout_sha")
       self.checkout_sha
+      self.setup_build_specific_environment_variables
+    end
+
+    def setup_build_specific_environment_variables
+      self.environment_variables_set = []
+
+      # We try to follow the existing formats
+      # https://wiki.jenkins.io/display/JENKINS/Building+a+software+project
+      env_mapping = {
+        BUILD_NUMBER: current_build_number,
+        JOB_NAME: self.project.project_name,
+        WORKSPACE: self.project.local_repo_path,
+        GIT_URL: self.repo.git_config.git_url,
+        GIT_SHA: self.current_build.sha,
+        BUILD_URL: "https://fastlane.ci", # TODO: actually build the URL, we don't know our own host, right?
+        CI_NAME: "fastlane.ci"
+      }
+
+      if self.git_fork_config && self.git_fork_config.branch.to_s.length > 0
+        env_mapping[:GIT_BRANCH] = self.git_fork_config.branch # TODO: does this work?
+      else
+        env_mapping[:GIT_BRANCH] = "master" # TODO: use actual default branch?
+      end
+
+      # We need to duplicate some ENV variables
+      env_mapping[:CI_BUILD_NUMBER] = env_mapping[:BUILD_NUMBER]
+      env_mapping[:CI_BUILD_URL] = env_mapping[:BUILD_URL]
+      env_mapping[:CI_BRANCH] = env_mapping[:GIT_BRANCH]
+      # env_mapping[:CI_PULL_REQUEST] = nil # TODO: do we have the PR information here?
+
+      env_mapping.each do |key, value|
+        set_build_specific_env_variable(key: key, value: value)
+      end
+
+      # TODO: to add potentially
+      # - BUILD_ID
+      # - BUILD_TAG
+    end
+
+    def set_build_specific_env_variable(key:, value:)
```

Comment:
this is dangerous because we're setting the environment for all builds that could be running. Currently it's not an issue, but if we ever do concurrent builds, this will cause race conditions.

Reply:
Cool, I see a few potential solutions:
1) If we use a child process, we can safely set the ENV variables just for the child process, with no way of the ENV variable surfacing up to the fastlane.ci main thread. This works very easily when using shell commands (which we don't)
2) The current implementation should work fine since our assumption is that one macOS hosts runs only one fastlane instance at the same time from what I understand. As soon as we implement distributed building (https://github.com/fastlane/ci/issues/457) we need a way to transfer the build metadata (git repo, branch, lane and build environment variables) anyway, so it's something we're gonna have to think about & implement once we take on #457 

Does that sound correct?